### PR TITLE
Add ChildrenInZoomRange method

### PR DESCRIPTION
**Pull request by peitili**
Add a utility method to generate all children tiles of a tile.

### CODE REVIEW

**Comment by peitili** [comment]
@paulmach would you please take a look?

File: maptile/tile.go [hunk 1]
```diff
@@ -226,6 +226,31 @@ func (t Tile) Children() Tiles {
...
+		panic("tile.Z is greater than zoomStart or zoomEnd")
+	}
+
+	zDeltaStart := zoomStart - tile.Z
+	zDeltaEnd := zoomEnd - tile.Z
+
+	res := make([]Tile, 0)
+
+	for d := zDeltaStart; d <= zDeltaEnd; d++ {
+		xTopLeft := tile.X << d
```
Comment (paulmach) [inline maptile/tile.go:241]: this variable seems more like `left` or `xStart`

Then the next one is more like `top` or `yStart`

File: maptile/tile.go [hunk 2]
```diff
@@ -226,6 +226,31 @@ func (t Tile) Children() Tiles {
...
+		yTopLeft := tile.Y << d
+		dim := uint32(math.Pow(2, float64(d)))
```
[the remark this answers is not in the document]
Comment (peitili) [reply maptile/tile.go]: Good point, fixed in next commit.

File: maptile/tile.go (hunk 1 above)
Comment (peitili) [reply maptile/tile.go]: Fixed in next commit.

File: maptile/tile.go [hunk 3]
```diff
@@ -226,6 +226,31 @@ func (t Tile) Children() Tiles {
 	}
 }
 
+// ChildrenInZoomRange returns all the children tiles of tile from ranges [zoomStart, zoomEnd], both ends inclusive.
+func ChildrenInZoomRange(tile Tile, zoomStart, zoomEnd Zoom) Tiles {
+	if zoomStart < tile.Z || zoomEnd < tile.Z {
```
[the remark this answers is not in the document]
Comment (peitili) [reply maptile/tile.go]: Fixed in the next commit.

**Pull request merged**